src/Three.d.ts

```diff
@@ -50,15 +50,12 @@ export * from './loaders/Loader';
 export * from './loaders/LoaderUtils';
 export * from './loaders/Cache';
 export * from './loaders/AudioLoader';
-export * from './lights/SpotLightShadow';
 export * from './lights/SpotLight';
 export * from './lights/PointLight';
 export * from './lights/RectAreaLight';
 export * from './lights/HemisphereLight';
-export * from './lights/DirectionalLightShadow';
 export * from './lights/DirectionalLight';
 export * from './lights/AmbientLight';
-export * from './lights/LightShadow';
```

Comment:
These are part of the public API, even if they aren't constructed by users. For example, TS users will want type information when they do:

```ts
light.shadow.mapSize.width = 512;
```

You could make a case for accessing the constructors too, if I want to check `light.shadow instanceof SpotLightShadow` or something, but I don't feel so strongly about that.

Reply:
Okay. Um, but that means TS users don't get type information for `PointLightShadow` right now. How about adding `PointLightShadow` to `Three.d.ts` and just remove the JS exports? Not sure if `light.shadow instanceof SpotLightShadow` is a compelling use case.

Reply:
Sounds good to me. 👍

Reply:
Updated 👍 .